components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp

```diff
@@ -70,127 +70,128 @@
             }
         }
-        }
 
-        if ("true".equals(request.getParameter("enableAudienceRestriction"))) {
-
-        	String audiencesCountParameter = SAMLSSOUIUtil.getSafeInput(request, "audiencePropertyCounter");
-        	if (audiencesCountParameter != null && !"".equals(audiencesCountParameter)) {
-            	try {
-                	int audiencesCount = Integer.parseInt(audiencesCountParameter);
-                	for (int i = 0; i < audiencesCount; i++) {
-                    	String audience = SAMLSSOUIUtil.getSafeInput(request, "audiencePropertyName" + i);
-                    	if (audience != null && !"".equals(audience) && !"null".equals(audience)) {
-                        	String[] currentAudiences = serviceProviderDTO.getRequestedAudiences();
-                        	boolean isAudienceAlreadyAdded = false;
-                        	for (String currentAudience : currentAudiences) {
-                            	if (audience.equals(currentAudience)) {
-                               		isAudienceAlreadyAdded = true;
-                                	break;
-                            	}
-                        	}
-
-                        	if (!isAudienceAlreadyAdded) {
-                            	serviceProviderDTO.addRequestedAudiences(audience);
-                        	}
-                    	}
-                	}
-            	} catch (NumberFormatException e) {
-                	throw new RuntimeException("Invalid number", e);
-            	}
-        	}
+        if (Boolean.parseBoolean(request.getParameter(SAMLSSOUIConstants.ENABLE_AUDIENCE_RESTRICTION))) {
+
+            String audiencesCountParameter = SAMLSSOUIUtil.getSafeInput(request, "audiencePropertyCounter");
+            if (StringUtils.isNotEmpty(audiencesCountParameter)) {
+                try {
+                    int audiencesCount = Integer.parseInt(audiencesCountParameter);
+                    for (int i = 0; i < audiencesCount; i++) {
+                        String audience = SAMLSSOUIUtil.getSafeInput(request, "audiencePropertyName" + i);
+                        if (StringUtils.isNotEmpty(audience) && !"null".equals(audience)) {
+                            String[] currentAudiences = serviceProviderDTO.getRequestedAudiences();
+                            boolean isAudienceAlreadyAdded = false;
+                            for (String currentAudience : currentAudiences) {
+                                if (audience.equals(currentAudience)) {
+                                    isAudienceAlreadyAdded = true;
+                                    break;
+                                }
+                            }
+
+                            if (!isAudienceAlreadyAdded) {
+                                serviceProviderDTO.addRequestedAudiences(audience);
+                            }
+                        }
+                    }
+                } catch (NumberFormatException e) {
```

Comment:
Would you be able to explain the logic implemented in this particular line? A more specific "sub-classed" RuntimeException is caught and thrown a "generic" RuntimeException. Generally one would not quite throw generic RuntimeExceptions. On the other hand, what's caught appears to be what needs to be thrown to the upstream code rather than using generic RuntimeExceptions, therefore, what if we get rid of this particular logic?